docs/docsite/rst/roadmap/ROADMAP_2_5.rst

```diff
@@ -0,0 +1,218 @@
+- Option for galaxy to remove 'old roles' on install (upgrade?), this is not clear cut as version can be a commit SHA and order there is not related to sorting, clear 'versions 1.1, 1.2' can use loose versioning comparisons.
+- ansible-galaxy cli should also be able to change the 'base role name' to point to specific versions, this solves the issue when the 'latest' is not actually the one existing plays should be using, again this mimics the 'alternatives' functionality.
+
+Globalize Callbacks
+-------------------
+- Make send_callback available to other code that cannot use it.
+- Would allow for 'full formatting' of output (see JSON callback)
+- Fixes static 'include' display problem
+
+Runtime Check on Modules for Blacklisting
+-----------------------------------------
+- Filter on things like "supported_by" in module metadata
+- Provide users with an option of "warning, error or allow/ignore"
+- Configurable via ansible.cfg and environment variable
+
+Windows
+-------
+- Implement gather_subset on Windows facts 
+- Move setup.ps1 guts to module_utils to allow arbitrary modules to call/refresh individual facts.
+- Fix Windows binary module support to work properly with become/env/async in all cases.
+- Fix Windows async + become to allow them to work together
+- Solve Windows become/env support for raw/script
+- Implement Windows become flags for controlling various modes
+
+  - logontype
+  - elevation behavior
+  - Add dict support to become_flags.
+- Fix Windows auto-kinit with threaded workers (or disallow it)
+- Finish C#/Powershell module_utils rewrite, convert core modules to use it.
+- Convert win_updates to action plugin for auto reboot and extra features
+- Spike out support for Windows Nano Server
+- Spike out changing the connection over to PSRP instead of WSMV
+- Module updates
+
+  - win_updates
+  
+    - Fix win_updates to detect (or request) become
+    - Add whitelist/blacklist features to win_updates
+  - win_dsc further improvements
+
+General Cloud
+-------------
+- Make multi-cloud provisioning easier
+
+  - Document multi-instance provisioning with loop directive
+  - Extend async_status to accept a list or build new action to simplify the with_items/register/until:finish patterns.
+- Diff mode will output provisioning task results of ansible-playbook runs
+- Terraform module
+
+AWS
+---
+- Focus on pull requests for various modules
+- Triage existing merges for modules
+- Module work
+
+  - ec2_instance 
+  - ec2_vpc: Allow the addition of secondary IPv4 CIDRS to existing VPCs.
+  - AWS Network Load Balancer support (NLB module, ASG support, etc)
+  - rds_instance
+
+Azure
+-----
+- Azure CLI auth
+- Fix Azure module results to have "high-level" output instead of raw REST API dictionary
+- Deprecate Azure automatic storage accounts in azure_rm_virtualmachine
+
+Network Roadmap
+---------------
+- Refactor common network shared code into package
+- Convert various nxos modules to leverage declarative intent 
```

Comment:
Per network-meeting: 

> No changes will be done in 2.5, we may add some of the missing platform specific modules under the `net_` (platform agnostic) versions, which will include DI, though again, we are not going to change anything here  (gundalow, 16:12:19)


Has this changed? pinging @gundalow @privateip

Reply:
@calfonso can you speak to this also?

Reply:
@thaumos @itdependsnetworks we had already started on NXOS for 2.4, it's not a new platform we are adding. We ran out of time for 2.4, so picking up some of the stuff we didn't get done. We are not changing how DI is designed in 2.5.

Reply:
Can you explain this on the weekly network meeting? The whole issue about DI was it was not well defined or understood. It was agreed it would take to long to clearly document, but there is time to add to the featureset?

Reply:
@itdependsnetworks I'm not aware of the time constraint around documentation. The documentation discussion was about how we want to advertise DI modules and the direction we should have other people take when implementing modules. For the platforms we already implemented using DI, we are taking time to finish up some features we didn't get to. What else do you want to know?